### PR TITLE
fix: Improve error if super.setUp() not called

### PR DESCRIPTION
**Pull request by david-allison**
Adds in lint check and fixes error message

Previous Error was:

```
java.lang.UnsatisfiedLinkError: 'long net.ankiweb.rsdroid.NativeMethods.openBackend(byte[])'
	at net.ankiweb.rsdroid.NativeMethods.openBackend(Native Method)
	...
```

The `Error` is only thrown in the case that Robolectric is used

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Ah! https://github.com/ankidroid/Anki-Android/blob/6369461b52a0e7165f01779ab1a9a2e6537a59f1/lint-release.xml#L125

File: lint-release.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <issue id="NewApi" severity="fatal" />
     <issue id="HardcodedText" severity="fatal" />
     <issue id="InlinedApi" severity="fatal" />
+    <issue id="MissingSuperCall" severity="fatal" />
```
Comment (ShridharGoel) [inline lint-release.xml:26]: Would this not throw errors when even super call is not required?

Comment (david-allison) [reply lint-release.xml:26]: A method shouldn't have `@CallSuper` if it's not required

**Review by ShridharGoel — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
Nice! CallSuper annotation is literally one of my favorite things about Java. It doesn't exist in ECMAScript or Typescript and I miss it a lot there. Saved me many times.

**Pull request merged**